bigmachine.go

```diff
@@ -362,13 +366,51 @@ func (b *B) pprofIndex(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func (b *B) shutdownAllMachines(ctx context.Context, duration time.Duration) {
+	b.mu.Lock()
```

Comment:
i think you only need to hold b.mu for the machines here, right?  maybe just get a snapshot instead?

```machines := b.Machines()```

Reply:
Yeah, but that would just be shifting the race from one place to another... I think if we care about this, we should implement proper "draining" (i.e., all calls to Start fail after Shutdown has initiated) -- but I think we shouldn't care: I think it's preferable to leave it up to the user (in this case Bigslice) to ensure that no new machines are started after shutdown is initiated.